lib/Dialect/LWE/Conversions/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -358,6 +358,21 @@ struct ConvertRAdd : public OpConversionPattern<RAddOp> {
   }
 };
 
+struct ConvertRAddPlain : public OpConversionPattern<RAddOp_Plain> {
+  ConvertRAddPlain(mlir::MLIRContext *context)
+      : OpConversionPattern<RAddOp_Plain>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      RAddOp_Plain op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<::mlir::heir::polynomial::AddOp>(
+        op, adaptor.getOperands()[0], adaptor.getOperands()[1]);
```

Comment:
not sure why the other patterns use `getOperands()[0]` in this file, but the `OpAdaptor` should have named methods corresponding to the tablegen names (`getLhs()`, `getRhs()`) that you can use here.

Reply:
in an OpConversionPattern `OpAdaptor` is defined as `SourceOp::Adaptor` where `SourceOp` is the template argument (`RAddOp_Plain` here). Cf. https://github.com/llvm/llvm-project/blob/5dc3cd0ee40c00d9fb542488fa5a54ff70273112/mlir/include/mlir/Transforms/DialectConversion.h#L586

Reply:
Oh, I see - I guess I was thinking of the (now removed) OneToNConversionPattern case, where it's really just a GenericAdaptor... that *does* raise the question (unrelated to this PR) of why all the patterns here use the same style 🙈